packages/core/src/builtin-tasks/contract.ts

```diff
@@ -0,0 +1,209 @@
+import { execSync } from 'node:child_process'
+import path from 'node:path'
+import fs from 'node:fs'
+import TOML from '@iarna/toml'
+import { key } from '@ckb-lumos/hd'
+import { generateAddress } from '@ckb-lumos/helpers'
+import { predefined } from '@ckb-lumos/config-manager'
+import { Script } from '@ckb-lumos/base'
+import { task, subtask } from '../config/config-env'
+import { paramTypes } from '../params'
+import { getUserConfigPath } from '../project-structure'
+
+task('contract').setAction(async () => {
+  execSync('kuai contract --help', { stdio: 'inherit' })
+})
+
+subtask('contract:get-workspace').setAction(async (_, { config }) => {
+  if (config.contract?.workspace) {
+    return config.contract?.workspace
+  }
+
+  const userConfigPath = getUserConfigPath()
+  if (!userConfigPath) {
+    throw new Error('Please run in kuai project file')
+  }
+
+  return path.join(path.dirname(userConfigPath), 'contract')
+})
+
+subtask('contract:set-environment').setAction(async () => {
+  // todo: download ckb-cli & capsule etc...
+})
+
+interface ImportArgs {
+  privateKey: string
+}
+
+subtask('contract:import-private-key')
+  .addParam('privateKey', '', '', paramTypes.string, false)
+  .setAction(async ({ privateKey }: ImportArgs) => {
+    const lockArg = key.privateKeyToBlake160(privateKey)
+
+    const res = execSync('ckb-cli account list --output-format json').toString()
+    const accountList: Array<{ lock_arg: string }> = JSON.parse(res)
+
+    if (accountList.some((account) => account.lock_arg === lockArg)) {
+      return
+    }
+
+    fs.writeFileSync('./.tmp_pk', privateKey)
+
+    console.log('add the user to ckb-cli, enter a password for the signature please')
+    execSync('ckb-cli account import --privkey-path ./.tmp_pk', { stdio: 'inherit' })
+
+    fs.rmSync('./.tmp_pk')
```

Comment:
It should be wrapped by `try cache` and remove the temp file in the `finally` block to make sure it will be erased

Reply:
Besides, it's better to use a random file name to avoid predictable file reading.